packages/entities/entities-shared/src/composables/useExternalLinkCreator.ts

```diff
@@ -7,11 +7,12 @@ export default function useExternalLinkCreator(routeParams: string[]): string {
     const fullUrl = `${baseUrl}${path}`
     const hasEmptyParams = routeParams.some(item => item.trim() === '')
 
-    /* Test validity of generated URL */
-    const urlRegex = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g // eslint-disable-line no-useless-escape
+    // Test validity of generated URL; only allows `http` protocal if domain is `localhost`
+    /* eslint-disable-next-line no-useless-escape */
+    const validUrlRegex = /^(https:\/\/(www\.)?([a-zA-Z0-9-]+\.){1,}[a-zA-Z]{2,}(:[0-9]+)?(\/[^\/]+)*)$|^(https|http):\/\/localhost(:[0-9]+)?(\/[^\/]+)*$/
```

Comment:
Would it make sense to add a unit test for this composable to test happy and bad paths?

Reply:
Good call - pushed a commit with a few more failing cases